### PR TITLE
add logic for azure:select-subscription

### PR DESCRIPTION
**Pull request by andxu**

### CODE REVIEW

File: azure-maven-plugin/src/main/java/com/microsoft/azure/plugin/login/SelectSubscriptionMojo.java [hunk 1]
```diff
@@ -23,11 +35,102 @@
...
+                    }
+
+                }
+
+                final Scanner scanner = getScanner();
+                int userSelected = -1;
+                while (true) {
+                    System.out.printf("Enter index value for subscription id: ");
+                    System.out.flush();
+                    final String input = scanner.nextLine();
```
Comment (Flanker32) [inline azure-maven-plugin/src/main/java/com/microsoft/azure/plugin/login/SelectSubscriptionMojo.java]: `scanner.nextInt()`?

File: azure-maven-plugin/src/main/java/com/microsoft/azure/plugin/login/SelectSubscriptionMojo.java [hunk 2]
```diff
@@ -23,11 +35,102 @@
      * The maven cli argument for set the active subscription by id or name
      */
     @Parameter(property = "subscription")
-    private String subscription;
+    private String subscriptionId;
 
     @Override
-    public void execute() throws MojoExecutionException {
-        throw new NotImplementedException("Not implemented.");
+    public void execute() throws MojoExecutionException, MojoFailureException {
```
Comment (Flanker32) [inline azure-maven-plugin/src/main/java/com/microsoft/azure/plugin/login/SelectSubscriptionMojo.java]: Shall we split this method? Seems it contains a bit much operations,

File: azure-auth-helper/src/main/java/com/microsoft/azure/auth/AzureAuthHelper.java [hunk 3]
```diff
@@ -149,9 +158,108 @@ public static AzureCredential readAzureCredentials(File file) throws IOException
...
+
+    /**
+     * Get azure credential from AzureCredential instance.
+     *
+     * @param credentials the azure credential
+     * @param env the azure environment
+     * @return the azure token credential can be used in Azure SDK.
+     * @throws IOException when there are some IO errors.
+     */
+    public static AzureTokenCredentials getMavenAzureLoginCredentials(AzureCredential credentials, AzureEnvironment env) throws IOException {
```
Comment (Flanker32) [inline azure-auth-helper/src/main/java/com/microsoft/azure/auth/AzureAuthHelper.java]: @NotNull for credentials?

File: azure-maven-plugin/src/main/java/com/microsoft/azure/plugin/login/LoginMojo.java [hunk 4]
```diff
@@ -30,19 +38,39 @@
...
+            String previousSubscriptionId = null;
+            try {
+                previousSubscriptionId = AzureAuthHelper.existsAzureSecretFile() ?
+                        AzureAuthHelper.readAzureCredentials(AzureAuthHelper.getAzureSecretFile()).getDefaultSubscription() : null;
+            } catch (IOException e) {
+                // ignore;
+            }
+
+            try {
+                newAzureCredential = devicelogin  ? null : AzureAuthHelper.oAuthLogin(env);
```
Comment (Flanker32) [inline azure-maven-plugin/src/main/java/com/microsoft/azure/plugin/login/LoginMojo.java]: what if user execute device loging by `-DdeviceLogin?`

Comment (andxu) [reply azure-maven-plugin/src/main/java/com/microsoft/azure/plugin/login/LoginMojo.java]: good catch

File: azure-maven-plugin/src/main/java/com/microsoft/azure/plugin/login/SelectSubscriptionMojo.java (hunk 2 above)
Comment (andxu) [reply azure-maven-plugin/src/main/java/com/microsoft/azure/plugin/login/SelectSubscriptionMojo.java]: there is a TODO comment.

File: azure-maven-plugin/src/main/java/com/microsoft/azure/plugin/login/SelectSubscriptionMojo.java (hunk 1 above)
Comment (andxu) [reply azure-maven-plugin/src/main/java/com/microsoft/azure/plugin/login/SelectSubscriptionMojo.java]: https://stackoverflow.com/questions/13102045/scanner-is-skipping-nextline-after-using-next-or-nextfoo

nextInt will introduce strange codes

File: azure-auth-helper/src/main/java/com/microsoft/azure/auth/AzureAuthHelper.java (hunk 3 above)
Comment (andxu) [reply azure-auth-helper/src/main/java/com/microsoft/azure/auth/AzureAuthHelper.java]: @NotNull will have no actual check if the parameter is null?

**Pull request merged**